framework/Models/PostProcessor.py

```diff
@@ -218,31 +221,26 @@ def collectOutput(self,finishedJob,output,options=None):
         dictionary of options that can be passed in when the collect of the output is performed by another model (e.g. EnsembleModel)
       @ Out, None
     """
-    if output.type not in self.validDataType:
-      self.raiseAnError(IOError, 'Output type', str(output.type), 'is not allowed!')
     outputCheckInfo = (output.name, output.type)
     if outputCheckInfo in self.inputCheckInfo:
       self.raiseAnError(IOError, 'DataObject',output.name,'is used as both input and output of', \
               self.interface.printTag, 'This is not allowed! Please use different DataObjet as output')
 
-    evaluation = finishedJob.getEvaluation()
-    if isinstance(evaluation, Runners.Error):
-      self.raiseAnError(RuntimeError, "No available output to collect (run possibly not finished yet)")
-    outputRealization = evaluation[1]
-
-    if output.type in ['PointSet','HistorySet']:
-      if self.outputDataset:
-        self.raiseAnError(IOError, "DataSet output is required, but the provided type of DataObject is", output.type)
-      self.raiseADebug('Dumping output in data object named ' + output.name)
-      if self.outputMultipleRealizations:
-        if 'dims' in outputRealization:
-          dims = outputRealization['dims']
-        else:
-          dims = {}
-        print(outputRealization.keys())
-        output.load(outputRealization['data'], style='dict', dims=dims)
-      else:
-        output.addRealization(outputRealization)
-    elif output.type in ['DataSet']:
-      self.raiseADebug('Dumping output in DataSet named ' + output.name)
-      output.load(outputRealization, style='dataset')
+    self._pp.collectOutput(finishedJob, output)
+
+  def provideExpectedMetaKeys(self):
+     """
+       Overrides the base class method to assure child postprocessor is also polled for its keys.
+       @ In, None
+       @ Out, meta, tuple, (set(str),dict), expected keys (empty if none) and the indexes related to expected keys
+     """
+     # get keys as per base class
+     metaKeys,metaParams = super().provideExpectedMetaKeys()
+     # add postprocessor keys
+     try:
+       keys, params = self._pp.provideExpectedMetaKeys()
+       metaKeys = metaKeys.union(keys)
+       metaParams.update(params)
+     except AttributeError:
+       pass # either "interface" has no method for returning meta keys, or "interface" is not established yet.
```

Comment:
this should not be possible. If it is in the base class, this AttributeError should not ever happen.

Reply:
updated. I have removed the try-except.